### ISSUE TITLE
Strange training behaviour using categorical_crossentropy on output from softmax

### ISSUE THREAD

**Issue by ziky90** [fchollet/keras#6983]
I'm using `keras==2.04` and `tensorflow==1.1.0` and I'm experiencing following problem.

When I am plying with custom ResNet50 for semantic segmentation that ends by the softmax activation that goes straight to the `losses.categorical_crossentropy()` function, then following things are  happening to me (multiple times on different types of datasets):
![image](https://user-images.githubusercontent.com/1810839/27129889-3fe6f8c0-5104-11e7-8906-448ae1909c3a.png)
![image](https://user-images.githubusercontent.com/1810839/27129904-4b4a054a-5104-11e7-86ac-2473af1c0db6.png)
![image](https://user-images.githubusercontent.com/1810839/27129918-57ad847e-5104-11e7-966b-0ace35e97f02.png)

I have managed to fix this thing by 2 different approaches:

1. ~~This seems to me rather as a hack, use clipvalue for gradient => but the training does not goes well (loss and accuracy are still pretty much the same)~~
2. **Have "raw" logits as the output from the ResNet50 and use basically else condition from `losses. categorical_crossentropy()`, so `tf.nn.softmax_cross_entropy_with_logits()`, here everything seems to be working as expected**

eg.
experiment on dataset1:
![screen shot 2017-06-14 at 09 50 34](https://user-images.githubusercontent.com/1810839/27130665-8a5813d2-5107-11e7-8b34-81e5a31b63c8.png)
![screen shot 2017-06-14 at 09 50 49](https://user-images.githubusercontent.com/1810839/27130678-92dbaa50-5107-11e7-82a6-18bee68e4996.png)
![screen shot 2017-06-14 at 09 51 02](https://user-images.githubusercontent.com/1810839/27130684-98a25a7e-5107-11e7-93b5-bf14ba4466a4.png)

experiment on dataset2:
![screen shot 2017-06-14 at 13 40 55](https://user-images.githubusercontent.com/1810839/27130687-9f56b7e8-5107-11e7-812c-540943e49322.png)
![screen shot 2017-06-14 at 13 41 07](https://user-images.githubusercontent.com/1810839/27130694-a43ec9b2-5107-11e7-9cce-fd4d487f3570.png)
![screen shot 2017-06-14 at 13 41 18](https://user-images.githubusercontent.com/1810839/27130701-aa9af358-5107-11e7-9a53-413dea6f50b9.png)

I assume that there is some bug somewhere here https://github.com/fchollet/keras/blob/master/keras/backend/tensorflow_backend.py#L2745 I'd expect that there might be performed division by 0 here or loss becomes negative at some time?
